ui/app/AppLayouts/Communities/views/PermissionsView.qml

```diff
@@ -44,16 +44,32 @@ ColumnLayout {
     signal userRestrictionsToggled(bool checked)
 
     readonly property alias count: repeater.count
-    ListModel {
-        id: communityItemModel
 
-        Component.onCompleted: {
-            append({
-                       text: root.communityDetails.name,
-                       imageSource: root.communityDetails.image,
-                       color: root.communityDetails.color
-                   })
+    Connections {
+        target: root.communityDetails
```

Comment:
Here we assume taht `communityDetails` is a `QtObject` (not e.g. js object) so we could change:
```
required property var communityDetails
```
to
```
required property QtObject communityDetails
```
to ensure that `name`/`image`,`color` are notifiable

Reply:
@marcioatdias I think your code is correct and totally needed!

@micieslak has a good point. It would be nice to have type safety at property level on top of your changes, to promote the `communityDetails` from `var` to `QtObject`. Now, with your changes, we're using it in the `Connections.target` and it expects a `QtObject`.

Thank you for your contribution 👍